### PR TITLE
replace nullptr with 0

### PR DESCRIPTION
**Pull request by sailfish009**

### CODE REVIEW

File: torch/csrc/Device.cpp [hunk 1]
```diff
@@ -181,39 +181,39 @@ PyTypeObject THPDeviceType = {
...
-  nullptr,                                     /* tp_dealloc */
-  nullptr,                                     /* tp_print */
-  nullptr,                                     /* tp_getattr */
-  nullptr,                                     /* tp_setattr */
-  nullptr,                                     /* tp_reserved */
+  0,                                     /* tp_dealloc */
+  0,                                     /* tp_print */
+  0,                                     /* tp_getattr */
+  0,                                     /* tp_setattr */
+  0,                                     /* tp_reserved */
```
Comment (vishwakftw) [inline torch/csrc/Device.cpp]: We only need to change:
```
nullptr, /* tp_print */
```
to
```
0  /* tp_vectorcall_offset */
```
for Python 3.8 compatibility.

**Review by vishwakftw — CHANGES_REQUESTED** [review]
We don't have Python 3.8 in the CI yet, so there's no way to test if this change is indeed the way to go.

Also cc: @ezyang

**Comment by albanD** [comment]
Hi,

What is the reason for this change? Our code base is c++11.

File: torch/csrc/Device.cpp (hunk 1 above)
Comment (ezyang) [reply torch/csrc/Device.cpp]: Yeah, I don't think it's appropriate to change everything to zero. Let's just do the localized change. However, I won't block this PR on Python 3.8 CI existing.

**Comment by vishwakftw** [comment]
@sailfish009 would you like to make the suggested changes?

**Comment by sailfish009** [comment]
@vishwakftw yes. thanks.

**Comment by vishwakftw** [comment]
This has been superseded by #29302

**Pull request closed without merging**